src/Element_TEST.cc

```diff
@@ -745,19 +765,46 @@ TEST(Element, DocRightPane)
 /////////////////////////////////////////////////
 TEST(Element, SetEmpty)
 {
-  sdf::Element elem;
+  std::stringstream buffer;
+  sdf::testing::RedirectConsoleStream redir(
+      sdf::Console::Instance()->GetMsgStream(), &buffer);
+
+  #ifdef _WIN32
+    sdf::Console::Instance()->SetQuiet(false);
+    sdf::testing::ScopeExit revertSetQuiet(
+      []
+      {
+        sdf::Console::Instance()->SetQuiet(true);
+      });
+  #endif
+
+  sdf::ElementPtr elem = std::make_shared<sdf::Element>();
+  sdf::Errors errors;
+
+  ASSERT_FALSE(elem->Set<std::string>(errors, ""));
+  ASSERT_EQ(errors.size(), 0u);
+
+  elem->AddValue("int", "0", true, errors, "value");
+  ASSERT_EQ(errors.size(), 0u);
+  ASSERT_TRUE(elem->Set<std::string>(errors, ""));
```

Comment:
Great catch! Indeed this could be a bug fix that changes behavior as it seems it's been like that before. Maybe we should log this change separately on the change log? is it easier with a separate PR for this?

I went ahead and fixed it here for now.

Reply:
Since this is a bug in the code, as per video call, I opened a separate PR to keep better track of this fix [here](https://github.com/gazebosim/sdformat/pull/1256).

Reply:
Sounds good. Thanks!